### PR TITLE
Create DHCP server to serve the jobs subnet

### PR DESCRIPTION
**Pull request by emdantrim**
## What is the problem that this PR is trying to fix?
We want to use a DHCP server that is not pfSense to serve the jobs subnet.

## What approach did you choose and why?
Creating a DHCP server using Terraform.

## How can you test this?
Apply it! (Or check out how I've already applied it)

## What feedback would you like, if any?
Are all the variables for the DHCP configuration acceptable?

### CODE REVIEW

File: modules/macstadium_infrastructure/dhcp-server.tf [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    ipv4_address       = "${cidrhost(var.jobs_network_subnet, 10 + var.index)}"
+    ipv4_prefix_length = "18"
+  }
+
+  disk {
+    template  = "${vsphere_folder.vanilla_vms.path}/${var.vanilla_image}"
+    datastore = "${var.datastore}"
+  }
+
+  connection {
```
Comment (meatballhat) [inline modules/macstadium_infrastructure/dhcp-server.tf]: Is this `connection` used?

File: modules/macstadium_infrastructure/dhcp-server.tf [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    agent = true
+  }
+}
+
+resource "null_resource" "dhcp_server" {
+  triggers {
+    host_id = "${vsphere_virtual_machine.dhcp_server.uuid}"
+  }
+
+  connection {
```
Comment (meatballhat) [inline modules/macstadium_infrastructure/dhcp-server.tf]: Same here; is this `connection` used?

File: modules/macstadium_infrastructure/dhcp-server.tf (hunk 1 above)
Comment (emdantrim) [reply modules/macstadium_infrastructure/dhcp-server.tf]: yep! this connection is used by the `macstadium_dhcp` module's `remote-exec`.

File: modules/macstadium_infrastructure/dhcp-server.tf (hunk 2 above)
Comment (emdantrim) [reply modules/macstadium_infrastructure/dhcp-server.tf]: I... think? This is copied from the `macstadium_go_worker` module, and I assumed it was put in there for a good reason. :/

**Comment by emdantrim** [comment]
Yep, it works!

File: modules/macstadium_infrastructure/dhcp-server.tf (hunk 1 above)
Comment (meatballhat) [reply modules/macstadium_infrastructure/dhcp-server.tf]: How? :grin:  I thought `macstadium_dhcp` was having ssh user and host injected as variables.

File: modules/macstadium_infrastructure/dhcp-server.tf (hunk 2 above)
Comment (meatballhat) [reply modules/macstadium_infrastructure/dhcp-server.tf]: I don't see anything using it, so ✂️ 🎉 !

Comment (meatballhat) [reply modules/macstadium_infrastructure/dhcp-server.tf]: Come to think of it, I don't see what this `resource "null_resource"` is doing at all 😁

**Pull request merged**